tests/test_cli_download_files.py

```diff
@@ -27,3 +27,51 @@ def test_download_files():
     assert test_result.output.endswith("\nDownload completed!\n")
     if os.path.isfile(test_file):
         os.remove(test_file)
+
+
+def test_download_files_filter_name():
+    """Test download_files() command with filter-name options."""
+    test_file = "3005/0d0714743f0204ed3c0144941e6ce248.configFile.py"
+    if os.path.isfile(test_file):
+        os.remove(test_file)
+    test_download_files_filter = CliRunner()
+    test_result_name = test_download_files_filter.invoke(
+        download_files, ["--recid", 3005, "--filter-name", "configFile.py"]
+    )
+    assert test_result_name.exit_code == 0
+    assert os.path.isfile(test_file) is True
+    assert test_result_name.output.endswith("\nDownload completed!\n")
+    if os.path.isfile(test_file):
+        os.remove(test_file)
+
+
+def test_download_files_filter_regexp():
+    """Test download_files() command with filter-regexp options."""
+    test_file = "3005/0d0714743f0204ed3c0144941e6ce248.configFile.py"
+    if os.path.isfile(test_file):
+        os.remove(test_file)
+    test_download_files_filter = CliRunner()
+    test_result_regexp = test_download_files_filter.invoke(
+        download_files, ["--recid", 3005, "--filter-regexp", "config"]
+    )
+    assert test_result_regexp.exit_code == 0
+    assert os.path.isfile(test_file) is True
+    assert test_result_regexp.output.endswith("\nDownload completed!\n")
+    if os.path.isfile(test_file):
+        os.remove(test_file)
+
+
+def test_download_files_filter_range():
+    """Test download_files() command with filter-range options."""
+    test_file = "3005/0d0714743f0204ed3c0144941e6ce248.configFile.py"
+    if os.path.isfile(test_file):
+        os.remove(test_file)
+    test_download_files_filter = CliRunner()
+    test_result_regexp = test_download_files_filter.invoke(
+        download_files, ["--recid", 3005, "--filter-range", "0-1"]
```

Comment:
I'm not sure users would have a tendency to start enumerating from 0 as in Python?  Especially if we are printing "Downloading file 1 of 10..."  Perhaps change the logic so that CLI option is passed starting from 1.

Reply:
... also, note that range 0-0 works, 0-1 works, but 1-1 does not do anything:

```console
$ cernopendata-client download-files --recid 3005 --filter-range 1-1

Download completed!
```

Reply:
It means when user mentions `1`, the file with `0` index will be considered as first file?

Reply:
> It means when user mentions `1`, the file with `0` index will be considered as first file?

Yes, exactly.  Basically, when a user asks for `n`-th file, you do `files[n-1]` internally...

Reply:
The range-starting-from-1 is still not working after latest update.  Here's the use case:

```console
$ cernopendata-client download-files --recid 5500 | head -4
==> Downloading file 1 of 11)
==> Downloading file: ./5500/BuildFile.xml
==> Downloading file 2 of 11
==> Downloading file: ./5500/HiggsDemoAnalyzer.cc
$ cernopendata-client download-files --recid 5500 --filter-range 1
==> Downloading file 1 of 1
==> Downloading file: ./5500/HiggsDemoAnalyzer.cc
Downloading: 81/81 kiB (100%)
Download completed!
```

Note that 2nd file was downloaded, not 1st...